src/storage/src/hummock/event_handler/uploader/mod.rs

```diff
@@ -1414,28 +1435,7 @@ impl HummockUploader {
         let UploaderState::Working(data) = &mut self.state else {
             return;
         };
-        if let Some(removed_table_data) = data.unsync_data.may_destroy_instance(instance_id) {
-            data.task_manager.remove_table_spill_tasks(
```

Comment:
For "drop streaming job", I think it is still preferable to cancel the task instead of waiting the task to finsih because the data will be deleted anyway. Can you elabrate in which cases we should preserve the upload task instead of cancelling it on instance destruction?

Reply:
Just updated the PR description. 

In short, this PR is not for `drop streaming job`, but for the single barrier configuration change in https://github.com/risingwavelabs/risingwave/issues/18312. In configuration change that may concurrently stop some existing actors and create new actors, for actors to be dropped, let's say in epoch1, they should write epoch1 data to hummock, and their data should be included in the SSTs of epoch1. But in the current main branch, for these actor to be dropped, their data is not likely to be included in the SSTs of epoch1, because their data might be dropped immediately when the instance is destroyed but before `sync(epoch1)` is called. For the same reason, we should also preserve the spilling task.

The current main branch works because in configuration change, we have a pause barrier, let's say epoch0, and we can ensure that no data is written between epoch0 and epoch1, and therefore even the instances are dropped, we won't lose data.

For normal `drop streaming job`, it's fine to drop instance and cancel related spilling task and ignore data loss. But given that we cannot distinguish whether dropping an instance is caused by drop streaming job or configuration change, we may let it follow the same handling logic as configuration change. I think the benefit of uploading less data may not be worthy compared to the increased complexity. Besides, a spilling task usually mix data from different tables, which may include normal table and the table to be dropped. Therefore, the spilling task may not be really cancelled when we call the previous `remove_table_spill_tasks`.

Reply:
>  a spilling task usually mix data from different tables, which may include normal table and the table to be dropped.

This is a good point. Thanks for the explanation.